app/src/test/java/com/example/trainingintive/viewmodels/DogImageViewModelTest.kt

```diff
@@ -0,0 +1,49 @@
+package com.example.trainingintive.viewmodels
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.Observer
+import com.example.trainingintive.TestSchedulersProvider
+import com.example.trainingintive.model.DogImageUrl
+import com.example.trainingintive.repository.DogImageRepository
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import io.reactivex.rxjava3.core.Completable
+import io.reactivex.rxjava3.core.Single
+import org.amshove.kluent.`should be equal to`
+import org.junit.Rule
+import org.junit.Test
+
+class DogImageViewModelTest {
+
+    @get:Rule
+    var instantTaskExecutorRule = InstantTaskExecutorRule()
```

Comment:
1. Why it is var not val? 2. It will be better to extract it to some class to not write it in every test.

Reply:
I don't know how to do it. You mean I can create class with @get:Rule and then extends this class in my tests classes?

Reply:
Yes